src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3917,28 +3917,27 @@ void Compiler::fgDebugCheckBlockLinks()
         // If this is a switch, check that the tables are consistent.
         // Note that we don't call GetSwitchDescMap(), because it has the side-effect
         // of allocating it if it is not present.
-        if (block->KindIs(BBJ_SWITCH) && m_switchDescMap != nullptr)
+        if (block->KindIs(BBJ_SWITCH) && (m_switchDescMap != nullptr))
         {
             SwitchUniqueSuccSet uniqueSuccSet;
             if (m_switchDescMap->Lookup(block, &uniqueSuccSet))
             {
-                // Create a set with all the successors. Don't use BlockSet, so we don't need to worry
-                // about the BlockSet epoch.
-                BitVecTraits bitVecTraits(fgBBNumMax + 1, this);
+                // Create a set with all the successors.
+                BitVecTraits bitVecTraits(compBasicBlockID, this);
                 BitVec       succBlocks(BitVecOps::MakeEmpty(&bitVecTraits));
                 for (BasicBlock* const bTarget : block->SwitchTargets())
                 {
-                    BitVecOps::AddElemD(&bitVecTraits, succBlocks, bTarget->bbNum);
+                    BitVecOps::AddElemD(&bitVecTraits, succBlocks, bTarget->bbID);
                 }
                 // Now we should have a set of unique successors that matches what's in the switchMap.
                 // First, check the number of entries, then make sure all the blocks in uniqueSuccSet
-                // are in the BlockSet.
+                // are in the BitVec.
                 unsigned count = BitVecOps::Count(&bitVecTraits, succBlocks);
                 assert(uniqueSuccSet.numDistinctSuccs == count);
                 for (unsigned i = 0; i < uniqueSuccSet.numDistinctSuccs; i++)
                 {
                     assert(BitVecOps::IsMember(&bitVecTraits, succBlocks,
-                                               uniqueSuccSet.nonDuplicates[i]->getDestinationBlock()->bbNum));
+                                               uniqueSuccSet.nonDuplicates[i]->getDestinationBlock()->bbID));
                 }
```

Comment:
Why this change?

Reply:
It's not necessary for removing `BlockSet`, but these are the last few usages of `bbNum`-based BitVecs, so I thought I'd get rid of them while I'm here.